src/bgw/job_stat.c

```diff
@@ -191,25 +201,36 @@ calculate_next_start_on_failure(TimestampTz last_finish, int consecutive_failure
 	float8 jitter = calculate_jitter_percent();
 	/* consecutive failures includes this failure */
 	float8 multiplier = 1 << (consecutive_failures - 1);
+	Timestamp res;
+	PG_TRY();
+	{
+		/* ival = retry_period * 2^(consecutive_failures - 1)  */
+		Datum ival = DirectFunctionCall2(interval_mul,
+										 IntervalPGetDatum(&job->fd.retry_period),
+										 Float8GetDatum(multiplier));
 
-	/* ival = retry_period * 2^(consecutive_failures - 1)  */
-	Datum ival = DirectFunctionCall2(interval_mul,
-									 IntervalPGetDatum(&job->fd.retry_period),
-									 Float8GetDatum(multiplier));
-
-	/* ival_max is the ceiling = MAX_INTERVALS_BACKOFF * schedule_interval */
-	Datum ival_max = DirectFunctionCall2(interval_mul,
-										 IntervalPGetDatum(&job->fd.schedule_interval),
-										 Float8GetDatum(MAX_INTERVALS_BACKOFF));
+		/* ival_max is the ceiling = MAX_INTERVALS_BACKOFF * schedule_interval */
+		Datum ival_max = DirectFunctionCall2(interval_mul,
+											 IntervalPGetDatum(&job->fd.schedule_interval),
+											 Float8GetDatum(MAX_INTERVALS_BACKOFF));
 
-	if (DatumGetInt32(DirectFunctionCall2(interval_cmp, ival, ival_max)) > 0)
-		ival = ival_max;
+		if (DatumGetInt32(DirectFunctionCall2(interval_cmp, ival, ival_max)) > 0)
+			ival = ival_max;
 
-	/* Add some random jitter to prevent stampeding-herds, interval will be within about +-13% */
-	ival = DirectFunctionCall2(interval_mul, ival, Float8GetDatum(1.0 + jitter));
+		/* Add some random jitter to prevent stampeding-herds, interval will be within about +-13%
+		 */
+		ival = DirectFunctionCall2(interval_mul, ival, Float8GetDatum(1.0 + jitter));
 
-	return DatumGetTimestampTz(
-		DirectFunctionCall2(timestamptz_pl_interval, TimestampTzGetDatum(last_finish), ival));
+		res = DatumGetTimestampTz(
+			DirectFunctionCall2(timestamptz_pl_interval, TimestampTzGetDatum(last_finish), ival));
+	}
+	PG_CATCH();
+	{
+		FlushErrorState();
+		res = ts_timer_get_current_timestamp();
```

Comment:
As in the previous case, rescheduling the job to run immediately does not sound like a good idea since it is likely to run into the same issue again. In addition, if there is some other issue that is causing the errors, it will cause the process to suddenly start to waste CPU. Better to report an error and not reschedule the job at all since that will allow a DBA to investigate the issue.

Reply:
we introduce some possible errors with the introduction of jitter calculation. (This is how I ran into the issue). Josh and I discussed this and thought a good compromise is probably to use "now() + retry_period" assuming we have a good value for retry_period .

( Note that we get into this error path only if the calculations fail in the earlier step - so this is not the default path. In general,  the job is started based on retry period and jitter computation. But we need a way of handling out-of-bound timestamps).

Reply:
Ok. Just skipping the current run and reporting an error sounds like a stable approach. This will not increase the CPU usage and cause other issues.  But *please* make sure to write an error message to the log so that DBAs can see that something is not working.

Reply:
we have to return a valid start time. This error is triggered usually by the internal calculations raising exceptions. This needs to be handled by the code as it is not a user triggered issue.

Reply:
Yes, that is why the error message should be in the log. This is where scrapers and DBAs are looking to figure out if anything is not working.